### DISCUSSION TITLE
Potential issue with accented characters and US Internationnal as input method

### DISCUSSION THREAD

**Discussion by artophe** [monkeytypegame/monkeytype#2403] [Questions] [answered]
Hi there!

I would like to report a potential issue with accented characters and US International as input method.

I’m using monkeytype on macOS with Firefox and Safari as web browsers, but I have the same issue on Chrome and multiple keyboards.
My input method is US International and  French as system language. This method allows me to type French accented characters. For example, when I need to form the character 'è', I type ` then e.

Unfortunately, when I need to type 'é' or 'ç' (respectively by typing ' then e and ' then c), the ' is detected as an error.

I don’t have this issue on Windows with US International as input method. Also, I don’t have this issue for the accented characters 'è', 'à' (\` then e and ` then a), ä, ï, ë, ü (" then a, " then i, etc.) on macOS.

I think I may have figured out why an input error is detected : monkeytype waits for the character ´ instead of ' when the user need to type 'é', 'ç', etc. Of course, on macOS and US International input method, it’s possible to enter the character ´ with Right Alt + E. But that means that a single character needs 3 inputs from the user.

Maybe a way to solve this issue would be to allow the character ' before the accented characters.

I tried monkeytype.com with Spanish and Portuguese which also have accented characters. Unfortunately, the same issue appears.

How can I help to solve this issue ? My English is awful and I’m not a developer.

Thanks everyone!

**Comment by Miodec** [comment, marked as answer]
The issue is that different systems handle characters like that differently. It works on Windows because it doesnt actually input the ' character, where as macOS does, and then pressing another key modifies the previous one. 

Even browsers can't settle on which key code is being sent when a dead key is pressed. Some send a 229, some don't. Android keyboards always send 229 for some reason.

Making it work with every language, every layout, every platform, every operating system is truly a nightmare.

Either way this issue is known, but no great solution has been found yet.

**Reply by artophe** [reply]
Hi! Thanks for your answer. I'll try to find a workaround for my setup.
